pika/adapters/base_connection.py

```diff
@@ -239,7 +241,10 @@ def _get_error_code(self, error_value):
         return None
 
     def _flush_outbound(self):
-        """Call the state manager who will figure out that we need to write."""
+        """write early, if the socket will take the data why not get it out
+        there asap.
+        """
+        self._handle_write()
```

Comment:
Yeah, you could do, though I'd be surprised if that overhead would really be noticeable compared to everything else we're doing in python. Also if we haven't dropped back into the ioloop in a while the kernel may have cleared some space in the buffer in the meantime, and in cases where we're consuming a full queue of small messages we can go quite a long time before dropping into the ioloop (it's what lead me to take a look at this in the first place). So I'd prefer to keep it as is, might not be the absolutely most efficient but it will keep the output buffer full and avoid stalling the pipeline.